text/0000-process-stdio-redirection.md

```diff
@@ -0,0 +1,93 @@
+- Feature Name: process-stdio-redirection
+- Start Date: 2015-04-10
+- RFC PR:
+- Rust Issue:
+
+# Summary
+
+Update the `std::process` API with the ability to redirect stdio of child
+processes to any opened file handle.
+
+# Motivation
+
+The current API in `std::process` allows to either pipe stdio between parent and
+child process or redirect stdio to `/dev/null`. It would also be largely useful
+to allow stdio redirection to any currently opened `std::fs::File` (henceforth
+`File`) handle. This would allow redirecting stdio with a physical file or even
+another process (via OS pipe) without forcing the parent process to buffer the
+data itself.
+
+For example, one may wish to spawn a process which prints gigabytes
+of data (e.g. logs) and use another process to filter through it, and save the
+result to a file. The current API would force the parent process to dedicate a
+thread just to buffer all data between child processes and disk, which is
+impractical given that the OS can stream the data for us via pipes and file
+redirection.
+
+# Detailed design
+
+First, the standard library should provide an OS agnostic way of creating OS
+in-memory pipes, i.e. a `Pipe`, providing reader and writer handles as `File`s.
+This would avoid the need for users to write OS specific (and `unsafe`) code
+while retaining all the benefits offered by `File`. This proposal considers
+making the `Pipe`'s reader and writer fields public to allow easily moving
+ownership of the two handles, but any other appropriate interface is acceptable.
+
+```rust
+pub struct Pipe {
+	pub reader: File,
```

Comment:
Could you add some details on how this structure is going to be created as well? (the more detailed an RFC is the better!)

Also, I think that `File` may not be the best type to use here, or at least it may depend on the implementation. I would guess that on unix this would use the `pipe` system call, but `File` has methods like `set_len` which may not work out too well for pipes.

As a final point, could you add some words as to how this will be implemented on Windows as well?